### PR TITLE
Filters viewmodel

### PR DESCRIPTION
**Pull request by DeeaMatei**
:gem: Create Filter ViewModel
#### Description
Moved some data managing logic from the FreaksFragment to the FilterViewModel

#### https://trello.com/c/PfS1t5Kn/153-android-freaks-screen-filters-viewmodel

Todo:

- [x] use binding for reset
- [x] use same pattern for projects as we use for skills
- [x] test reset (skill and projects)
- [x] filters list not fully showing on landscape

### CODE REVIEW

**Comment by DeeaMatei** [comment]
[Android - Freaks Screen - Filters ViewModel](https://trello.com/c/PfS1t5Kn/153-android-freaks-screen-filters-viewmodel)

File: features/freaks/src/main/kotlin/com/agilefreaks.freaks_catalog.features.freaks/FilterViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package com.agilefreaks.freaks_catalog.features.freaks
+
+import androidx.lifecycle.ViewModel
+
+class FilterViewModel : ViewModel() {
+    companion object {
```
Comment (AlexandraDamaschin) [inline features/freaks/src/main/kotlin/com/agilefreaks.freaks_catalog.features.freaks/FilterViewModel.kt]: There seems to be some guidance about where we should put out companion objects in a class. Did you read about this? Check this out and let us know what you think about this. 
https://kotlinlang.org/docs/coding-conventions.html#class-layout

**Review by AlexandraDamaschin — CHANGES_REQUESTED** [review]
Small improvement

File: features/freaks/src/main/kotlin/com/agilefreaks.freaks_catalog.features.freaks/FilterViewModel.kt (hunk 1 above)
Comment (DeeaMatei) [reply features/freaks/src/main/kotlin/com/agilefreaks.freaks_catalog.features.freaks/FilterViewModel.kt]: I haven't read about this before. I think it's nice to have a set of rules in order to keep things clean, organized and easier to read.

**Comment by cipry13** [comment]
https://projects.invisionapp.com/prototype/Freaks-Catalogue-ckr6ckqz30013hz013qsnnq52/play/fd7f781c

Reset button design should be changed.
Divider should be on top of each item, not at the bottom.

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,5 +1,12 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
+  <component name="DesignSurface">
+    <option name="filePathToZoomLevelMap">
+      <map>
+        <entry key="features/freaks/src/main/res/layout/bottom_sheet_dialog.xml" value="0.2015625" />
+      </map>
+    </option>
+  </component>
```
Comment (cipry13) [inline .idea/misc.xml]: this shouldn't be here

File: buildSrc/src/main/kotlin/commons/android-library.gradle.kts [hunk 3]
```diff
@@ -83,6 +83,7 @@ dependencies {
 
     implementation(Dependencies.Koin.CORE)
     implementation(Dependencies.Koin.ANDROID)
+//    implementation(Dependencies.Koin.ANDROID_VIEWMODEL)
```
Comment (cipry13) [inline buildSrc/src/main/kotlin/commons/android-library.gradle.kts]: this line should be removed

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    fun reset() {
+        skills.value?.forEach { it.reset() }
+        projects.value?.forEach { it.reset() }
+    }
+
+    fun applyFilters(): List<String> =
+        listOf("", "")
+
+    private fun loadSkills(): List<String> =
```
Comment (cipry13) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt:34]: you shouldn't leave data used for testing

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreaksFragment.kt [hunk 5]
```diff
@@ -86,52 +92,44 @@ class FreaksFragment : Fragment() {
...
-        } else {
-            listOf("Freaks Catalog", "Proj2", "Tutorial", "Altkeva")
-        }
+    private fun showFilterModal(list: LiveData<List<FilterItem>>, name: String) {
+        val dialog = setupFilterModal(name)
+        list.observe(viewLifecycleOwner, {
+            val adapter = FilterAdapter()
+            adapter.submitList(it)
+            val recyclerFiltersView = dialog.findViewById<RecyclerView>(R.id.recycler_filters_view)
+            recyclerFiltersView.adapter = adapter
```
Comment (deniscoman) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreaksFragment.kt]: This should be part of the setup filter

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreaksFragment.kt [hunk 6]
```diff
@@ -86,52 +92,44 @@ class FreaksFragment : Fragment() {
...
+        dialog.setCancelable(true)
 
-        btApply?.setOnClickListener {
+        val recyclerFiltersView =
+            mBottomSheetBinding.recyclerFiltersView
+        recyclerFiltersView.layoutManager = LinearLayoutManager(requireContext())
 
-        }
+        dialog.setContentView(mBottomSheetBinding.root)
+        dialog.show()
```
Comment (deniscoman) [inline features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/FreaksFragment.kt]: And only this and state should be part of the show filter method

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt (hunk 4 above)
Comment (deniscoman) [reply features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt:34]: @cipry13 why is this data used for testing?

**Review by deniscoman — CHANGES_REQUESTED** [review]

File: features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt (hunk 4 above)
Comment (cipry13) [reply features/freaks/src/main/kotlin/com/agilefreaks/freaks_catalog/features/freaks/model/FilterViewModel.kt:34]: @deniscoman the loadSkills() method had duplicated values when I left the comment.

**Pull request merged**